### DISCUSSION TITLE
Simpler API for rules/shortcuts

### DISCUSSION THREAD

**Discussion by TheComputerM** [unocss/unocss#471] [Ideas]
The *shortcuts* api is a bit redundant as you can easily do the things you do there in *rules*.
```js
shortcuts: {'btn': 'px-4'}
// to
rules: ['btn', {'padding-left': '1rem', 'padding-right': '1rem'}]
```

I propose to remove shortcuts and add and new key to rules:
```js
rules: ['btn', {'@apply': 'px-4', '&::before': {content: "''", '@apply': 'bg-red-500'}}]
```

### Benefits
- Can make components/utilities with both `@apply` and manual CSS.
- Easy to compose components with dark mode in mind.
- Less confusing to new comers.
- Can make shortcuts using obscure CSS properties that have no utility.

**Comment by TheComputerM** [comment]
If the end goal of unocss is still to be merged into windi then ignore this as windicss already has this feature.

**Comment by XEngine** [comment]
I definitly support this idea as this also makes plugin migration from tailwind to unocss easier
